lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -427,18 +428,24 @@ private function generateMagicGet(ClassMetadata $class)
             if ($reflectionClass->getMethod('__get')->returnsReference()) {
                 $returnReference = '& ';
             }
-        }
+
+			$nameParameterTypeHint = $this->getMethodParameterTypeHintByReflection($reflectionClass, '__get', 'name');
+			$returnTypeHint = $this->getMethodReturnType($reflectionClass->getMethod('__get'));
+        } else {
+			$nameParameterTypeHint = null;
```

Comment:
Please set these to a default before the `if` block: `else` block can then be omitted